core/controllers/base_test.py

```diff
@@ -107,9 +120,55 @@ def test_redirect_in_both_logged_in_and_logged_out_states(self):
         self.login('user@example.com')
         response = self.testapp.get('/')
         self.assertEqual(response.status_int, 302)
+
+        # Homepage is either dashboard or library based on user's learner
+        # status, this homepage redirect rule behavior is seperately tested
+        atHome = ('dashboard' in response.headers['location'] or
+                  'library' in response.headers['location'])
+        self.assertTrue(atHome, msg="user must be at library or dashboard")
+        self.logout()
+
+    def test_root_redirect_rules_for_learners(self):
+        # Test learner behavior. Login as learner
+        self.login(self.TEST_LEARNER_EMAIL)
+
+        # Dont create any exploration and simply go to root
+        response = self.testapp.get('/')
+        self.assertIn('library', response.headers['location'])
+        self.logout()
+
+    def test_root_redirect_for_creators(self):
+        # Test creator behavior. Login as creator
+        self.login(self.TEST_CREATOR_EMAIL)
+        creator_user_id = self.get_user_id_from_email(self.TEST_CREATOR_EMAIL)
+        exploration_id = '0_en_test_exploration'
+
+        # Create a test exploration
+        self.save_new_valid_exploration(
+            exploration_id, creator_user_id, title='Test',
+            category='Test', language_code='en')
+
+        # go to root url
```

Comment:
Here and elsewhere: for actions that are obvious from the code, you don't need a comment to describe what is going on. Use comments to explain _why_ you're doing things if needed, but don't use them if they just repeat what the code is obviously doing.